File: frontend/src/app/models/Vehicle.ts


export interface Vehicle {
    id: string;
    model: string;
    brand: string;
    yearOfProduction: number;
    imageUrl: string;
}

export interface VehicleDetails {
    id: string;
    model: string;
    brand: string;
    yearOfProduction: number;
    imageUrl?: string;
    fuel: string;
    seats: number;
    description: string;
    status: string;
    department: string;
    departmentId: string;
    price: number;
}

export interface VehicleEditValues {
    yearOfProduction: number;
    seats: number;
    price: number;
    description: string;
    imageUrl?: string;
}

export interface VehicleCreateValues {
    description: string;
    imageUrl?: string;
    registration: string;
    vin: string;
    seats: number;
    yearOfProduction: number;
    price: number;
    modelId: string;
    departmentId: string;
    fuelId: string;
}

export interface VehicleForRent {
    id: string;
    model: string;
    brand: string;
    yearOfProduction: number;
}

export interface VehicleFilters {
    fuels: Fuel[];
    brands: Brand[];
    departments: Department[];
}

interface Fuel {
    id: string;
    type: string;
}

interface Brand {
    id: string;
    name: string;
    models: Model[]
}

interface Model {
    id: string;
    name: string;
}

interface Department {
    id: string;
    name: string;
}

export interface SelectedFilters {
    departmentId?: string;
    fuelId?: string;
    brandId?: string;
    modelId?: string;
    seats: number
    minPrice: number;
    maxPrice: number;
}

File: backend/Domain/Models/Vehicle.cs
namespace Domain.Models;

public class Vehicle
{
    public Guid Id { get; set; }
    public string Description { get; set; } = default!;
    public string? ImageUrl { get; set; }
    public string Registration { get; set; } = default!;
    public string Vin { get; set; } = default!;
    public int Seats { get; set; }
    public int YearOfProduction { get; set; }
    public Guid VehicleStatusId { get; set; }
    public Guid DepartmentId { get; set; }
    public Guid ModelId { get; set; } 
    public Guid PriceId { get; set; }
    public Guid FuelId { get; set; }

    public Fuel Fuel { get; set; } = default!;
    public Price Price { get; set; } = default!;
    public Model Model { get; set; } = default!;
    public Department Department { get; set; } = default!;
    public VehicleStatus VehicleStatus { get; set; } = default!;
    public ICollection<Rent> Rentings { get; set; } = default!;
}